coding-concepts.js

```diff
@@ -6,49 +6,106 @@
+// const colors = ["tangerine", "magenta", "lilac", "daffodil"]
 // console.log(colors.push("indigo"))
 
-// a) Your answer:
+
+// a) Your answer: adds "indigo" to the end of the array.
 // b) Verify and explain:
 
+//      My guess:
+
+
+//      The variable colors that is equal to the array with all of the color strings incased in the brackets[].
+
+//      The .push() built-in method adds a value to the end of the array.
+
+//      The value in the array is the string "indigo".
+
+//      So i presume the code will display the the colors in the array with the addition of "indigo" at the end of the array.
+
+//      Incorrect.
+
+//      Correction:
+
+//      To get the answer i explained above, I'd have to add another console.log with the colors variable as the argument.
+
+//      the correct answer was 5 for the total number of strings in the projected code.
+
+//
+
 // --------------------1) What will this log?
 
-const cohort = "LEARN 2023"
+// const cohort = "LEARN 2023"
 // console.log(cohort.length)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: 10
+// b) Verify and explain: 
+
+//      My guess:
+
+//      I say its going to be 10. The .length function call count in an Index format and it starts at 0 instead on 1. 
+//      There is 11 characters in the string so it will the character from 0 to 10. So the answer is 10.
 
 // --------------------2) What will this log?
 
-const greeting = "Hello World!"
+// const greeting = "Hello World!"
 // console.log(greeting[4])
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: o
+// b) Verify and explain: 
+
+        // My guess:
+
+        // The console.log function call has the variable in the argument with the number 4 in the brackets. The variable
+        // reference the string "Hello World". The 4 will count the string in an Index format from 0 to 4 so the answer should be "o"
 
 // --------------------3) What will this log?
 
-const languages = ["JavaScript", "Ruby", "Python", "C++"]
-const index = 1
+// const languages = ["JavaScript", "Ruby", "Python", "C++"]
+// const index = 1
 // console.log(languages[index])
 
-// a) Your answer:
+// a) Your answer: Ruby
 // b) Verify and explain:
 
+//  My guess: 
+
+//  We declare the variable "languages" equal to the array.
+
+//  In the argument, we have the index variable that is equal to the value (1).
+
+//  The array is counting the string from an index format from 0 to 1. 
+
+//  The answer should be "Ruby"
+
 // --------------------4) What will this log?
 
-const weekendDays = ["saturday", "sunday"]
+// const weekendDays = ["saturday", "sunday"]
 // console.log(weekendDays.toUpperCase())
 
-// a) Your answer:
+// a) Your answer: SATURDAY, SUNDAY
 // b) Verify and explain:
 
+//  My guess:
+
+//  I think that it will display in all caps. the variable in the fuction call console.log() had both strings in the array
+//  with the .toUpperCase() built-in method that capitalize the elements in the array.
```

Comment:
We get a TypeError. Since .toUpperCase only works on strings, if you try to use it on an array like this, it will produce an error.